2D_platformer_template/platformer_stock.py

```diff
@@ -0,0 +1,897 @@
+
+            if lcollide_id != -1 and self.square_render.colliderect(
+                    object_list[lcollide_id]) and \
+                    self.left_col.colliderect(object_list[lcollide_id]) and \
+                    collide_y < self.ypos + self.height and \
+                    self.ypos < collide_y + collide_height:
+
+                if self.xpos < collide_x + collide_width and \
+                        not self.top_col.colliderect(object_list[lcollide_id]):
+                    self.xpos = collide_x + collide_width
+
+        all_xr = []
+        # Right side collision, going right to turn left
+        for rcollide_id in right_collision:
+            collide_x = object_list[rcollide_id].x
+            collide_y = object_list[rcollide_id].y
+            collide_height = object_list[rcollide_id].height
+
+            if self.right_col.colliderect(object_list[rcollide_id]) and \
+                    not self.top_col.colliderect(object_list[rcollide_id]) and \
+                    not self.bot_col.colliderect(object_list[rcollide_id]) and \
+                    self.xpos + self.width < collide_x:
+                all_xr += [collide_x]
+
+            if rcollide_id != -1 and self.square_render.colliderect(
+                    object_list[rcollide_id]) and \
+                    self.right_col.colliderect(object_list[rcollide_id]) and \
+                    collide_y < self.ypos + self.height and \
+                    self.ypos < collide_y + collide_height:
+
+                if collide_x < self.xpos + self.width and \
+                        not self.top_col.colliderect(object_list[rcollide_id]):
+                    self.xpos = collide_x - self.width
+
+        if 0 < len(all_xl):
+            self.left_x = max(all_xl)
+        else:
+            self.left_x = None
+        if 0 < len(all_xr):
+            self.right_x = min(all_xr)
+        else:
+            self.right_x = None
+
+    def gravity(self):
+        # fix turning off gravity
+        if self.enable_gravity and not self.jump_ability:
+            gravity_y = ((self.gravity_counter ** 2) *
+                         0.00015) * self.res_height
+        else:
+            gravity_y = 0
+
+        if self.enable_gravity and not self.jump_ability:
+            if self.grav_y is not None and \
+                    self.grav_y < gravity_y + self.ypos + self.height:
+                self.ypos = self.grav_y - self.height
+                self.enable_gravity = False
+                self.jump_ability = True
+                self.gravity_counter = self.max_gravity
+            else:
+                self.ypos += gravity_y
+
+        if self.gravity_counter < 1100:
+            self.gravity_counter += 2
+
+    def death(self, death_list: [pygame.Rect]):
+        collide_id = self.square_render.collidelist(death_list)
+        if collide_id != -1:
+            self.alive = False
+            return 1
+        else:
+            return 0
+
+
+class Program:

```

Comment:
Since an old game stock template was used, memory is also missing (and the width/height scaling).